lib/ansible/module_utils/facts/system/ssh_pub_keys.py

```diff
@@ -31,22 +31,32 @@ class SshPubKeyFactCollector(BaseFactCollector):
 
     def collect(self, module=None, collected_facts=None):
         ssh_pub_key_facts = {}
-        keytypes = ('dsa', 'rsa', 'ecdsa', 'ed25519')
+        algos = ('dsa', 'rsa', 'ecdsa', 'ed25519')
 
         # list of directories to check for ssh keys
         # used in the order listed here, the first one with keys is used
         keydirs = ['/etc/ssh', '/etc/openssh', '/etc']
 
         for keydir in keydirs:
-            for type_ in keytypes:
-                factname = 'ssh_host_key_%s_public' % type_
+            for algo in algos:
+                factname = 'ssh_host_key_%s_public' % algo
                 if factname in ssh_pub_key_facts:
                     # a previous keydir was already successful, stop looking
                     # for keys
                     return ssh_pub_key_facts
-                key_filename = '%s/ssh_host_%s_key.pub' % (keydir, type_)
+                key_filename = '%s/ssh_host_%s_key.pub' % (keydir, algo)
                 keydata = get_file_content(key_filename)
                 if keydata is not None:
                     ssh_pub_key_facts[factname] = keydata.split()[1]
+                    # Not guaranteed to work with protocol 1, which is deprecated
+                    keyfields = keydata.split()
+                    if len(keyfields) == 4:
```

Comment:
I completely agree. I couldn't find anything explicit about the host key file format in the man pages; I found analogous info in the AuthorizedKeysFile section and used that. I wanted to try to avoid breaking the fact on the off-chance there was an option present -- but if it's not checked, that's way better. Thanks for reporting the OpenSSH issue. I'll get the PR updated.